pkg/kubectl/cmd/apply.go

```diff
@@ -345,7 +345,11 @@ func RunApply(f cmdutil.Factory, cmd *cobra.Command, out, errOut io.Writer, opti
 
 			if string(patchBytes) == "{}" {
 				count++
-				cmdutil.PrintSuccess(shortOutput, out, info.Object, false, "unchanged")
+
+				// Do not print unchanged message if an output format other than "name" is specified
+				if len(output) == 0 || shortOutput {
+					cmdutil.PrintSuccess(shortOutput, out, info.Object, false, "unchanged")
+				}
 				return nil
```

Comment:
If it's unchanged, the function will just return here, without printing anything. 

I believe the intention of this PR is to print the object even when nothing is changed, without the "unchanged" message?

Reply:
The intention is to not print the "unchanged" message, since it mangles the expected YAML output. I don't particularly mind it not printing anything at all, as that is still "valid" YAML